src/db.rs

```diff
@@ -38,315 +39,338 @@ impl Default for Connection {
+    let dbpool = get_dbpool().await?;
 
     let keypair_b64 = base64::encode(&keypair);
-    let mqstr = "INSERT INTO report_keypair (key_id, keypair, polid) VALUES(?, ?, ?)";
 
-    trnsx.exec_drop(mqstr, (&id, &keypair_b64, &policy_id))?;
-    trnsx.commit()?;
+    let query = match policy_id {
+        Some(p) => format!(
+            "INSERT INTO report_keypair (key_id, keypair, polid) VALUES(\'{}\', \'{}\', {:?})",
+            id, keypair_b64, p
+        ),
+        None => format!(
+            "INSERT INTO report_keypair (key_id, keypair) VALUES(\'{}\', \'{}\')",
+            id, keypair_b64
+        ),
+    };
+
+    sqlx::query(query.as_str()).execute(&dbpool).await?;
 
     Ok(())
 }
 
-pub fn get_report_keypair(id: &str) -> Result<Vec<u8>> {
-    let mut dbconn = get_dbconn()?;
+pub async fn get_report_keypair(id: &str) -> Result<Vec<u8>> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "SELECT keypair FROM report_keypair WHERE key_id = ?";
+    let query = format!(
+        "SELECT keypair FROM report_keypair WHERE key_id = \'{}\'",
+        id
+    );
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-    let keys: Option<String> = trnsx.exec_first(mqstr, (id,))?;
-    let kp = keys.ok_or_else(|| anyhow!("report signing key not found"))?;
+    let keys_row = sqlx::query(query.as_str()).fetch_one(&dbpool).await?;
 
+    let kp = keys_row.try_get::<String, _>(0)?;
     let kp_bytes = base64::decode(&kp)?;
 
     Ok(kp_bytes)
 }
 
-pub fn delete_report_keypair(key_id: &str) -> Result<()> {
-    let mut dbconn = get_dbconn()?;
+pub async fn delete_report_keypair(key_id: &str) -> Result<()> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "DELETE FROM report_keypair WHERE key_id = ?";
+    let query = format!("DELETE FROM report_keypair WHERE key_id = \'{}\'", key_id);
 
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    sqlx::query(query.as_str()).execute(&dbpool).await?;
 
-    trnsx.exec_drop(mqstr, (key_id,))?;
-    trnsx.commit()?;
     Ok(())
 }
 
-pub fn get_signing_keys_policy(key_id: &str) -> Result<Option<policy::Policy>> {
-    let mut dbconn = get_dbconn()?;
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+pub async fn get_signing_keys_policy(key_id: &str) -> Result<Option<policy::Policy>> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "SELECT polid FROM report_keypair WHERE key_id = ? AND polid IS NOT NULL";
+    let query = format!(
+        "SELECT polid FROM report_keypair WHERE key_id = \'{}\' AND polid IS NOT NULL",
+        key_id
+    );
 
-    let policy_id: Option<u64> = trnsx.exec_first(mqstr, (key_id,))?;
+    let policy_id_row = sqlx::query(query.as_str()).fetch_one(&dbpool).await?;
 
-    if let Some(id) = policy_id {
-        Ok(Some(get_policy(id)?))
-    } else {
-        Ok(None)
-    }
+    let policy_id = policy_id_row.try_get::<i32, _>(0)? as u64;
+    Ok(Some(get_policy(policy_id).await?))
```

Comment:
I think you change the function semantics here. In case the query returns no rows, we want the function to return `Ok(None)`. However due to the question mark in `policy_id_row.try_get::<i32, _>(0)?`, I think it will return Err in the new implementation.

If I'm right, then we're missing a unit-test to verify this behavior.

Reply:
My updates for sqlx did change the semantics.  I updated to catch errors in the policy_id_row.try_get::<i32, ->(0) statement and also added a unit test.